tests/mir_test_framework/test_server.cpp

```diff
@@ -210,13 +169,30 @@ void miral::TestDisplayServer::invoke_tools(std::function<void(WindowManagerTool
     tools.invoke_under_lock([&]{f(tools); });
 }
 
-void miral::TestDisplayServer::invoke_window_manager(std::function<void(mir::shell::WindowManager& wm)> const& f)
+void TestDisplayServer::add_server_init(std::function<void(mir::Server&)>&& init)
 {
-    if (auto const wm = window_manager.lock())
-        f(*wm);
-    else
-        BOOST_THROW_EXCEPTION(std::runtime_error{"Server not running"});
+    auto temp = [old_init=init_server, new_init=std::move(init)](mir::Server& server)
+        {
+            old_init(server);
+            new_init(server);
```

Comment:
This is a little strange, but I suppose it works just as well as pushing to a vector of functions.